drivers/modbus_rtu/modbus_rtu.c

```diff
@@ -0,0 +1,469 @@
+
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+
+#include "xtimer.h"
+#include "thread.h"
+#include "periph/gpio.h"
+#include "msg.h"
+#include "checksum/ucrc16.h"
+#include "mutex.h"
+
+enum MESSAGE {
+  ID = 0,  //!< ID field
+  FUNC,    //!< Function code position
+  ADD_HI,  //!< Address high byte
+  ADD_LO,  //!< Address low byte
+  NB_HI,   //!< Number of coils or registers high byte
+  NB_LO,   //!< Number of coils or registers low byte
+  BYTE_CNT //!< byte counter
+};
+
+// bytes with crc16
+#define MIN_SIZE_REQUEST (8)
+
+#define lowByte(w) ((uint8_t)((w)&0xff))
+#define highByte(w) ((uint8_t)((w) >> 8))
+
+#define calcCRC(buffer, size) (ucrc16_calc_le(buffer, size, 0xA001, 0xFFFF))
+
+static inline void enamdle_trasmit(modbus_rtu_t *modbus);
+static inline void disamdle_trasmit(modbus_rtu_t *modbus);
+static inline void send(modbus_rtu_t *modbus);
+static inline int prepare_request(modbus_rtu_t *modbus);
+static inline int get_response(modbus_rtu_t *modbus);
+static void rx_cb(void *arg, uint8_t data);
+
+static inline int wait_bytes(modbus_rtu_t *modbus, uint8_t num) {
+  msg_t msg;
+  while (modbus->size_buffer < num) {
+    if (xtimer_msg_receive_timeout(&msg, modbus->rx_timeout) < 0) {
+      return -1;
+    }
+  }
+  return 0;
+}
+// from buffer
+static inline void read_address(modbus_rtu_t *modbus) {
+  modbus->msg->addr = ((modbus->buffer[ADD_HI]) << 8) | modbus->buffer[ADD_LO];
+}
+// to buffer
+static inline void write_address(modbus_rtu_t *modbus) {
+  modbus->buffer[ADD_HI] = highByte(modbus->msg->addr);
+  modbus->buffer[ADD_LO] = lowByte(modbus->msg->addr);
+}
+// from buffer
+static inline void read_count(modbus_rtu_t *modbus) {
+  modbus->msg->count = ((modbus->buffer[NB_HI]) << 8) | modbus->buffer[NB_LO];
+}
+// to buffer
+static inline void write_count(modbus_rtu_t *modbus) {
+  modbus->buffer[NB_HI] = highByte(modbus->msg->count);
+  modbus->buffer[NB_LO] = lowByte(modbus->msg->count);
+}
+
+static void copy_bits(uint8_t *dst, uint16_t start_bit_dst,
+                      const uint8_t *src, uint16_t start_bit_src, uint16_t count) {
+  while (count > 0) {
+    div_t s = div(start_bit_src, 8);
+    div_t d = div(start_bit_dst, 8);
+    // printf("dst byte: %d; bit: %d\n", d.quot, d.rem);
+    uint8_t bit = src[s.quot] & (1 << s.rem);
+    // printf("src byte: %d; bit: %d, val: %d\n", s.quot, s.rem, bit);
+    if (bit) {
+      dst[d.quot] |= 1 << (7 - d.rem);
+    } else {
+      dst[d.quot] &= ~(1 << (7 - d.rem));
+    }
+    start_bit_dst++;
+    start_bit_src++;
+    count--;
+  }
+}
+
+int modbus_rtu_init(modbus_rtu_t *modbus, uint32_t baudrate) {
+  int err = uart_init(modbus->uart, baudrate, rx_cb, modbus);
+  if (err != UART_OK) {
+    return err;
+  }
+  if (modbus->pin_rts != GPIO_UNDEF) {
+    gpio_init(modbus->pin_rts, GPIO_OUT);
+    gpio_write(modbus->pin_rts, !modbus->pin_rts_enable);
+  }
+  mutex_init(&(modbus->mutex_buffer));
+  // usec in sec / byte per second * 1.5
```

Comment:
Please use c comments
```suggestion
  /* usec in sec / byte per second * 1.5 */
```